src/Persistence_matrix/include/gudhi/Persistence_matrix/boundary_matrix.h

```diff
@@ -474,14 +474,16 @@ inline typename Boundary_matrix<Master_matrix>::index Boundary_matrix<Master_mat
 
   //updates container sizes
   if constexpr (Master_matrix::Option_list::has_row_access && !Master_matrix::Option_list::has_removable_rows) {
-    id_index pivot;
-    if constexpr (Master_matrix::Option_list::is_z2) {
-      pivot = *std::prev(boundary.end());
-    } else {
-      pivot = std::prev(boundary.end())->first;
+    if (boundary.size() != 0){
+      id_index pivot;
+      if constexpr (Master_matrix::Option_list::is_z2) {
+        pivot = *std::prev(boundary.end());
+      } else {
+        pivot = std::prev(boundary.end())->first;
+      }
+      //row container
+      if (ra_opt::rows_->size() <= pivot) ra_opt::rows_->resize(pivot + 1);
```

Comment:
You had a doubling strategy elsewhere, but here it isn't needed?

Reply:
I don't think I was very consistent with this strategy. There were also places where it was important that the vector had the exact right size. But I could probably use the doubling strategy here. Do you know if std::vector does already the doubling strategy by itself (that is, even if I resize by one, he reserves the double of size in case of reallocation?). If yes, do you think that doing it on our side is still useful?

Reply:
Yes, for a `resize` that requires increasing capacity, the standard library already has an exponential strategy (the constant is not always 2, but that's a detail). I checked with libstdc++, libc++, and msvc, and I think the standard requires it (resize is specified in terms of insertions at the end, and those are specified as having amortized constant complexity). So I guess there is no point doing anything subtle in our code.

(note that the situation would be completely different with `reserve`, where the library trusts that the user knows what they are doing, and does not second guess them)

Reply:
Ok, good to know!